File: src/Data/MemorySearchHandle.cpp
#include "MemorySearchHandle.h"

RedMemoryDump::MemorySearchHandle::MemorySearchHandle(const Red::CString& p_type,
                                                      uint64_t p_offset) : type(p_type), offset(p_offset) {
}

Red::CString RedMemoryDump::MemorySearchHandle::get_type() const {
    return type;
}

uint64_t RedMemoryDump::MemorySearchHandle::get_offset() const {
    return offset;
}

File: scripts/cet/Data/MemorySearchHandle.lua
---@class MemorySearchHandle
---@field type string
---@field offset number
local MemorySearchHandle = {}

---@param handles any[]
---@return MemorySearchHandle[]
function MemorySearchHandle.ToTable(handles)
  local results = {}

  for _, handle in ipairs(handles) do
    table.insert(results, {
      type = handle:GetType(),
      offset = handle:GetOffset(),
    })
  end
  return results
end

return MemorySearchHandle
